### PR TITLE
Feature/m2 2327 add variables support in reports

### PR DESCRIPTION
**Pull request by alex-i-e**

### CODE REVIEW

File: src/shared/utils/exportData/getJourneyCSVObject.ts [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
 
 const getTimeByCondition = (time: string) => (condition: boolean) => condition ? time : '';
 
-export const getJourneyCSVObject = (event: ExtendedEvent) => {
+export const getJourneyCSVObject = <T>({
+  event,
+  rawAnswersObject,
+}: {
+  event: ExtendedEvent;
+  rawAnswersObject: Record<string, T & { answer: AnswerDTO }>;
```
Comment (wildkiomi) [inline src/shared/utils/exportData/getJourneyCSVObject.ts:21]: is it a mistype? rowAnswersObject

File: src/shared/utils/exportData/replaceItemVariableWithName.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+const getDateString = (obj?: DecryptedDateAnswer['value']) => {
+  if (!obj) return '';
+
+  return `${obj.year}-${Number(obj.month + 1)
+    .toString()
+    .padStart(2, '0')}-${Number(obj.day).toString().padStart(2, '0')}`;
+};
+
+const reBrackets = /\[\[(.*?)]]/g;
```
Comment (wildkiomi) [inline src/shared/utils/exportData/replaceItemVariableWithName.ts:27]: rename please. why is it "re"?

File: src/shared/utils/exportData/replaceItemVariableWithName.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      const itemValue = itemsObject[variableName];
+      const answerValue = rawAnswersObject[variableName];
+      const rawAnswer = answerValue?.answer;
+
+      if (rawAnswer && Array.isArray((rawAnswer as DecryptedMultiSelectionAnswer).value)) {
+        const names: string[] = [];
+        (rawAnswer as DecryptedMultiSelectionAnswer).value.forEach((ans) => {
+          const item =
+            (itemValue.responseValues as SingleAndMultipleSelectItemResponseValues).options.find(
+              (option) => option.value === ans,
```
Comment (IhorNesterenko99) [inline src/shared/utils/exportData/replaceItemVariableWithName.ts:64]: pls rename ans and reg variables

**Review by IhorNesterenko99 — APPROVED** [review]

File: src/modules/Dashboard/hooks/useDecryptedActivityData.ts [hunk 4]
```diff
@@ -28,10 +28,10 @@ export const useDecryptedActivityData = (
   const { prime, base } = encryptionInfoFromServer;
   const privateKey = getAppletPrivateKey(dynamicAppletId ?? appletId);
 
-  return <T extends DecryptedAnswerSharedProps>(
+  return <T extends EncryptedAnswerSharedProps>(
     answersApiResponse: T,
   ): DecryptedActivityData<T> => {
-    const { userPublicKey, answer, items, itemIds, events, ...rest } = answersApiResponse;
+    const { userPublicKey, answer, itemIds, events, ...rest } = answersApiResponse;
```
Comment (wildkiomi) [inline src/modules/Dashboard/hooks/useDecryptedActivityData.ts:34]: add items

File: src/modules/Dashboard/hooks/useDecryptedActivityData.ts [hunk 5]
```diff
@@ -70,7 +70,7 @@ export const useDecryptedActivityData = (
       }
     }
 
-    const answerDataDecrypted = items.map((activityItem, index) => ({
+    const answerDataDecrypted = rest.items.map((activityItem, index) => ({
```
Comment (wildkiomi) [inline src/modules/Dashboard/hooks/useDecryptedActivityData.ts:73]: and here use just items

**Review by wildkiomi — APPROVED** [review]

**Pull request merged**